### DISCUSSION TITLE
How to change speed of agent traversing off-mesh-connection?

### DISCUSSION THREAD

**Discussion by rob-myers** [isaac-mason/recast-navigation-js#445] [Q&A] [answered]
Hey, I can detect when an agent is traversing a so-called off-mesh-connection via `agent.state() === 2`.

But how can I change its speed? 

Setting the parameters `maxSpeed` and `maxAcceleration` does not seem to work.

**Comment by rob-myers** [comment]
I suppose I could remove the agent, move it manually across link, then request the navPath again.

I could also try `agent.requestMoveVelocity`.

But it would be nice to do this without re-planning.

**Reply by isaac-mason** [reply, marked as answer]
Hey @rob-myers, unfortunately to the best of my knowledge recastnavigation currently doesn't have great support for customising traversal across off mesh links.

From reading recastnavigation google discussions and github issues, it seems most have to resort to modifying DetourCrowd code to add their own handling.

In the library the handling is a lerp between the two points, and it appears the lerp is time based, and only considers the agents max speed once when triggering moving over the offmesh connection.

Some relevant links to DetourCrowd code:
- https://github.com/isaac-mason/recastnavigation/blob/599fd0f023181c0a484df2a18cf1d75a3553852e/DetourCrowd/Source/DetourCrowd.cpp#L1164
- https://github.com/isaac-mason/recastnavigation/blob/599fd0f023181c0a484df2a18cf1d75a3553852e/DetourCrowd/Source/DetourCrowd.cpp#L1431 

> I suppose I could remove the agent, move it manually across link, then request the navPath again.

I have seen similar suggestions given to folks using recast. Unless you are going to go down the path of modifying recast, this may be worth trying!
